### PR TITLE
Export `calculateResiduals()` and `convertUnits()`

### LINKED ISSUE #1081: Turn on tests for `plotObservedVsSimulated()` on CI
Once this issue is resolved:
https://github.com/Open-Systems-Pharmacology/TLF-Library/issues/367

### LINKED ISSUE #1084: Export some plotting utility functions
Functions which have a broader utility.

- [ ] `ospsuite:::.calculateResiduals()`
- [ ] `ospsuite:::.unitConverter()`

If we decide to export the unit conversion function, I think it should be called `convertUnits()` (or something similar), since the verb should come first in the function name.

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
TODO:

- [x] Export `calculateResiduals()`
- [x] Export `convertUnits()`
- [x] Add docs for both of them
- [x] Update NEWS
- [x] Closes #1081 
- [x] Closes #1084

### CODE REVIEW

File: R/utilities-data-combined.R [hunk 1]
```diff
@@ -1,4 +1,344 @@
...
+  pairedData <- dplyr::tibble(
+    "xValues"      = observedData[, "xValues"],
+    "xUnit"        = unique(data$xUnit),
+    "xDimension"   = unique(data$xDimension),
+    "yValues"      = observedData[, "yValues"],
+    "yErrorValues" = yErrorValues,
+    "yUnit"        = unique(data$yUnit),
+    "yDimension"   = unique(data$yDimension),
+    "predValue"    = simulatedData[timeMatchedData, "yValues"]
+  )
```
Comment (Yuri05) [inline R/utilities-data-combined.R:228]: I wonder if it wouldn't be better for the understanding to rename the columns:
* X-Columns to Time-columns (e.g. `xValues`=>`timeValues `etc.) 
* Y-Columns to Observed-Columns (e.g. `yValues`=>`observedValues`)
* `predValue `=> `predictedValues `(in any case it  should be plural)

File: R/utilities-data-combined.R [hunk 2]
```diff
@@ -1,4 +1,344 @@
...
+
+  # The linear scaling is represented either of the following:
+  #
+  # - `"lin"` (in `DefaultPlotConfiguration`)
+  # - `"identity"` (in `tlf::PlotConfiguration`, because of `{ggplot2}`)
+  if (scaling %in% c("lin", "identity")) {
```
Comment (Yuri05) [inline R/utilities-data-combined.R:234]: why using strings instead of `tlf::Scaling$lin` etc.?

Comment (IndrajeetPatil) [reply R/utilities-data-combined.R:234]: Because `tlf::Scaling` doesn't have `"identity"` entry.

File: R/utilities-data-combined.R (hunk 1 above)
Comment (IndrajeetPatil) [reply R/utilities-data-combined.R:228]: I am just trying to be consistent with the naming schema present in the original data frame from which this paired data frame is extracted:

https://github.com/Open-Systems-Pharmacology/OSPSuite-R/blob/af89b3a242649124e077458ce29f37dd87516b72/R/data-combined.R#L378-L381

Yeah, I agree. Following names, although verbose, will be better:

- predValue -> predictedValues
- resValue -> residualValues

File: R/utilities-data-combined.R [hunk 3]
```diff
@@ -1,4 +1,344 @@
...
+#'   yValues = c(1.9, 6.1, 7, 8.2, 1, 0, 4, 8),
+#'   yErrorValues = rnorm(8),
+#'   yUnit = ospUnits$`Concentration [mass]`$`mg/l`,
+#'   yDimension = ospDimensions$`Concentration (mass)`
+#' )
+#'
+#' ospsuite:::.extractResidualsToTibble(df)
+#'
+#' @keywords internal
+.extractResidualsToTibble <- function(data, scaling = tlf::Scaling$lin) {
```
Comment (Yuri05) [inline R/utilities-data-combined.R:197]: I think the `scaling `argument should be **mandatory** here?

File: R/utilities-data-combined.R [hunk 4]
```diff
@@ -1,4 +1,344 @@
...
+  )
+
+  # The linear scaling is represented either of the following:
+  #
+  # - `"lin"` (in `DefaultPlotConfiguration`)
+  # - `"identity"` (in `tlf::PlotConfiguration`, because of `{ggplot2}`)
+  if (scaling %in% c("lin", "identity")) {
+    pairedData <- dplyr::mutate(pairedData, resValue = predValue - yValues)
+  } else {
+    pairedData <- dplyr::mutate(pairedData, resValue = log(predValue) - log(yValues))
```
Comment (Yuri05) [inline R/utilities-data-combined.R:237]: log(x) is ln(x) in R, right?
Seems wrong to me, I would expect `log10(predicted)-log10(observed)` and not ln(predicted)-ln(observed)???

Comment (IndrajeetPatil) [reply R/utilities-data-combined.R:237]: I am using the same calculations as in the reporting engine:

https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/blob/3a3aa9ac469643a0a50db34ca4b4aafdeae71250/R/utilities-goodness-of-fit.R#L189-L194

But, yes, you are right that `log(x)` in R is `log(x, exp(1))`.

File: R/utilities-data-combined.R (hunk 3 above)
Comment (IndrajeetPatil) [reply R/utilities-data-combined.R:197]: Good idea. Done!

File: R/utilities-data-combined.R (hunk 2 above)
Comment (IndrajeetPatil) [reply R/utilities-data-combined.R]: Oh, and I am using `tlf::Scaling$lin` only for that one entry now.

File: R/utilities-data-combined.R (hunk 1 above)
Comment (IndrajeetPatil) [reply R/utilities-data-combined.R:228]: Done.

File: R/utilities-data-combined.R (hunk 2 above)
Comment (Yuri05) [reply R/utilities-data-combined.R]: should we introduce `tlf::Scaling$identity` then?

File: R/utilities-data-combined.R (hunk 4 above)
Comment (Yuri05) [reply R/utilities-data-combined.R]: you are right, RE calculates it like this. interesting...

File: R/utilities-data-combined.R (hunk 3 above)
Comment (Yuri05) [reply R/utilities-data-combined.R:197]: hmm, it's still optional?

Comment (IndrajeetPatil) [reply R/utilities-data-combined.R:197]: Oh, this is internal function. I changed it in `calculateResiduals`, which is the exported function.

I guess I can also change it here.

File: R/utilities-data-combined.R (hunk 2 above)
Comment (PavelBal) [reply R/utilities-data-combined.R]: What kind of scaling is "Identity"?

File: R/utilities-data-combined.R (hunk 3 above)
Comment (IndrajeetPatil) [reply R/utilities-data-combined.R]: Done.

File: R/utilities-data-combined.R (hunk 2 above)
Comment (IndrajeetPatil) [reply R/utilities-data-combined.R]: Yes, good idea.

Created issue to track this: https://github.com/Open-Systems-Pharmacology/TLF-Library/issues/368

Comment (IndrajeetPatil) [reply R/utilities-data-combined.R]: `"identity"` is to `{ggplot2}` what `"lin"` is to `{tlf}`.

A linear scale is represented as `"identity"` in `tlf::PlotConfiguration$xAxis$scale`.

This is just for internal purposes. Users won't have to know about it.

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**